TESTING.md

```diff
@@ -0,0 +1,99 @@
+# tardigrade-ci testing documentation
+
+Testing tardigrade-ci implementation with various setup on local environment.
+
+## Testing
+
+This project can be utilized one of two ways, via docker or via a Makefile include.
+
+### Via Docker (non-tardigrade-ci image)
+
+In a non-tardigrade-ci container, the second condition in Makefile.bootstrap ought to fail, resulting in the auto-init logic creating the tardigrade-ci/ directory.
+
+1. Run the following bash script or manually create a test-project directory with the following configurations:
+
+```bash
+#!/bin/bash
+
+mkdir test-project
+cd test-project
+git init
+echo "FROM plus3it/tardigrade-ci:0.8.0" > Dockerfile
+echo "SHELL := /bin/bash\n" > Makefile
+echo 'include $(shell test -f .tardigrade-ci || curl -sSL -o .tardigrade-ci "https://raw.githubusercontent.com/plus3it/tardigrade-ci/master/bootstrap/Makefile.bootstrap"; echo .tardigrade-ci)' >> Makefile
+echo '# tardigrade-ci' > .gitignore
+echo '.tardigrade-ci' >> .gitignore
+echo 'tardigrade-ci/' >> .gitignore
+```
+
+2. Run the following docker command in the test-project directory:
+
+```bash
+docker run --rm -v $PWD:/workdir --workdir=/workdir --entrypoint make python:3 help
+```
+
+3. This should result in the creation of both .tardigrade-ci and /tardigrade-ci
+
+4. Run the following command to remove the tardigrade-ci subdirectory and bootstrap Makefile:
+
+```bash
+docker run --rm -v $PWD:/workdir --workdir=/workdir --entrypoint make python:3 clean
+```
+
+5. Go to parent directory of test-project and remove the directory for initial setup:
+
+```bash
+rm -rf test-project/
+```
+
+### Via Docker (tardigrade-ci image)
+
+Running a tardigrade-ci container should automatically put the user in developer mode and create only the .tardigrade-ci file in the local environment.
```

Comment:
> Running a tardigrade-ci container should automatically put the user in developer mode

This is not quite correct... When running the tardigrade-ci container, you get the version of tardigrade-ci from the container. "Developer mode" is when it uses tardigrade-ci/ from a sibling directory in your local environment, so as a tardigrade-ci developer you can make changes there, then test them quickly in another project. Maybe we need to note this as a third mode, "container mode"?

```suggestion
Running a tardigrade-ci container will use the version of `tardigrade-ci` present in the container, and so will create only the `.tardigrade-ci` file in the `test-project/` directory.
```